mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -177,32 +257,45 @@ object StringIndexer extends DefaultParamsReadable[StringIndexer] {
 /**
  * Model fitted by [[StringIndexer]].
  *
- * @param labels  Ordered list of labels, corresponding to indices to be assigned.
+ * @param labelsArray Array of ordered list of labels, corresponding to indices to be assigned
+ *                    for each input column.
  *
- * @note During transformation, if the input column does not exist,
- * `StringIndexerModel.transform` would return the input dataset unmodified.
+ * @note During transformation, if any input column does not exist,
+ * `StringIndexerModel.transform` would skip the input column.
+ * If all input columns do not exist, it returns the input dataset unmodified.
  * This is a temporary fix for the case when target labels do not exist during prediction.
  */
 @Since("1.4.0")
 class StringIndexerModel (
     @Since("1.4.0") override val uid: String,
-    @Since("1.5.0") val labels: Array[String])
+    @Since("3.0.0") val labelsArray: Array[Array[String]])
   extends Model[StringIndexerModel] with StringIndexerBase with MLWritable {
 
   import StringIndexerModel._
 
   @Since("1.5.0")
```

Comment:
For API compatibility we probably want to add a `def this(uid: String, labels: Array[String])` here -- although since it Spark 3 we could not if we don't think the old one is a useful entry point to continue to support. What are folks thoughts on that?

Reply:
My idea is that most of time users won't directly create `StringIndexerModel`. Either you load from a saved model or by fitting through `StringIndexer`. That's why I think we can skip that.

Reply:
That seems reasonable, we should call it out in the migration guide just in case though since it is a public API we are changing.